### ISSUE TITLE
Code highlight in web-ui?

### ISSUE THREAD

**Issue by melMass** [tree-sitter/tree-sitter#663]
Hi,
Is there a way currently or a plan to test highlight in the web-UI while developing a parser?
I have other questions I'll ask here because I don't want to open an issue for each but tell me if I should:

- Is there a kind of convention for the name of the rules (nodes)?
- Is there an API reference for the grammar functions? (repeat, alias, token ...)

Thanks

**Comment by maxbrunsfeld** [comment]
> Is there a way currently or a plan to test highlight in the web-UI while developing a parser?

There is not a way to do this right now, although you can insert your highlighting query into the `Query` textarea, which will just highlight each capture in your query using an arbitrary, unique color.

> Is there a kind of convention for the name of the rules (nodes)?

We haven't formalized anything yet, although there are a lot of informal similarities between the node in names in, for example, `tree-sitter-cpp`, `tree-sitter-javascript`, `tree-sitter-go`, and `tree-sitter-rust`.

> Is there an API reference for the grammar functions? (repeat, alias, token ...)

Right now, the closes thing to this is [this section](https://tree-sitter.github.io/tree-sitter/creating-parsers#the-grammar-dsl) of the docs.

**Comment by melMass** [comment]
Thanks Max, great job on TS.

> right now
Thanks I'll try `Query`, is this planned then? it could be useful as you now support WASM (next thing on my list is to look at tree-sitter-web)

> there are a lot of informal similarities between the node in names in

Yep first thing I did was to look at all the tree-sitter integrations.
I based mine on the C one (which is the closest to [VEX](https://www.sidefx.com/docs/houdini/vex/index.html)) , but the name used and the logic is quite hard to follow, so I'll look at other languages I know in the list.

> this section of the docs.

Thanks, yes the docs evolved from last time I checked sorry, I will go through all of it again.

Kind of unrelated but is there an official way to "bridge" tree-sitter & LSP?

**Comment by maxbrunsfeld** [comment]
There's no official way, but a number of people have built language servers using Tree-sitter, using different language bindings. Someone else asked about this recently here: https://github.com/tree-sitter/tree-sitter/issues/667#issuecomment-654331312.
